hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -1096,6 +1096,29 @@ threeNodesNoErrorsOnOpen tracer tmpDir node@RunningNode{nodeSocket} hydraScripts
       Right _headIsOpen ->
         pure ()
 
+-- | Hydra nodes ABC run and connect to each other.
+-- Hydra nodes ABC shut down.
+-- Hydra nodes AB run and connect to each other.
+-- Hydra nodes AB shut down.
+-- Hydra nodes ABC run and connect to each other.
+nodeCanSupportMultipleEtcdClusters :: Tracer IO EndToEndLog -> FilePath -> RunningNode -> [TxId] -> IO ()
+nodeCanSupportMultipleEtcdClusters tracer tmpDir RunningNode{nodeSocket} hydraScriptsTxId = do
+  aliceKeys <- generate genKeyPair
+  bobKeys <- generate genKeyPair
+  carolKeys <- generate genKeyPair
+
+  let cardanoKeysABC = [aliceKeys, bobKeys, carolKeys]
+      hydraKeysABC = [aliceSk, bobSk, carolSk]
+  let cardanoKeysAB = [aliceKeys, bobKeys]
+      hydraKeysAB = [aliceSk, bobSk]
+
+  let contestationPeriod = UnsafeContestationPeriod 2
+  let depositDeadline = UnsafeDepositDeadline 50
+  let hydraTracer = contramap FromHydraNode tracer
+  withHydraCluster hydraTracer tmpDir nodeSocket 1 cardanoKeysABC hydraKeysABC hydraScriptsTxId contestationPeriod depositDeadline $ \clients -> waitForNodesConnected hydraTracer 20 clients
+  withHydraCluster hydraTracer tmpDir nodeSocket 1 cardanoKeysAB hydraKeysAB hydraScriptsTxId contestationPeriod depositDeadline $ \clients -> waitForNodesConnected hydraTracer 20 clients
+  withHydraCluster hydraTracer tmpDir nodeSocket 1 cardanoKeysABC hydraKeysABC hydraScriptsTxId contestationPeriod depositDeadline $ \clients -> waitForNodesConnected hydraTracer 20 clients
```

Comment:
These are using always fresh directories.

Also, the problematic scenario is when one of `AB` is still running when the other reconfigures to `ABC`. Also `C` should just be able to keep running throughout and start working when both `A` and `B` have switched to `ABC`.